tests/test_bounce_lib.py

```diff
@@ -835,6 +861,29 @@ def test_crossover_bounce_mid_bounce_no_happy_old_lots_of_unhappy_old(self):
         assert actual['create_app'] is False
         assert len(actual['tasks_to_drain']) == 4
 
+    def test_crossover_bounce_using_margin_factor(self):
+        new_config = {'id': 'foo.bar.12345', 'instances': 500}
+        happy_tasks = [mock.Mock() for _ in xrange(100)]
+        old_app_live_happy_tasks = {
+            'app1': set(mock.Mock() for _ in xrange(300)),
+            'app2': set(),
+        }
+        old_app_live_unhappy_tasks = {
+            'app1': set(),
+            'app2': set(mock.Mock() for _ in xrange(200)),
+        }
+
+        actual = bounce_lib.crossover_bounce(
+            new_config=new_config,
+            new_app_running=True,
+            happy_new_tasks=happy_tasks,
+            old_app_live_happy_tasks=old_app_live_happy_tasks,
+            old_app_live_unhappy_tasks=old_app_live_unhappy_tasks,
+            margin_factor=0.95,
+        )
+        assert actual['create_app'] is False
+        assert len(actual['tasks_to_drain']) == 125
```

Comment:
Hm. I don't think this should drain anything, since there are only 400 healthy tasks and we've asked for 500 total. It should only drain anything if the number of healthy tasks is more than 475 = 0.95*500.